ludwig/utils/triton_utils.py

```diff
@@ -50,142 +98,620 @@ def forward(self, {input_signature}):
+    max_queue_delay_microseconds: Optional[int] = 100,
+) -> Dict[str, Tuple[str, str, int]]:
     """Exports a torchscript model to a output path that serves as a repository for Triton Inference Server.
 
     # Inputs
-
     :param model: (LudwigModel) A ludwig model.
+    :param data_example: (pd.DataFrame) an example from the dataset.
+        Used to get dimensions throughout the pipeline.
     :param output_path: (str) The output path for the model repository.
     :param model_name: (str) The optional model name.
-    :param model_name: (Union[int,str]) The optional model verison.
+    :param model_version: (Union[int,str]) The optional model verison.
+    :param preprocessor_num_instances: (int) number of instances for the preprocessor (on CPU).
+    :param predictor_device_type: (str) device type for the predictor to be deployed on. One of "cpu" or "cuda"
+    :param predictor_num_instances: (int) number of instances for the predictor.
+    :param postprocessor_num_instances: (int) number of instances for the postprocessor (on CPU).
+    :param predictor_max_batch_size: (int) max_batch_size parameter for the predictor Triton config.
+    :param max_queue_delay_microseconds: (int) max_queue_delay_microseconds for all Triton configs.
 
     # Return
-    :return: (str, str) The saved model path, and config path.
+    :return: (Dict[str, Tuple[str, str, int]]) The saved config path, model path, and model size for each of
+        the three inference modules.
     """
-    model_ts = generate_triton_torchscript(model)
-    model_dir = os.path.join(output_path, model_name, str(model_version))
-    os.makedirs(model_dir, exist_ok=True)
-    # Save the file to <model_repository>/<model_name>/<model_version>/model.pt
-    model_path = os.path.join(model_dir, "model.pt")
-    model_ts.save(model_path)
-    # Save the default onfig to <model_repository>/<model_name>/config.pbtxt
-    config_path = os.path.join(output_path, model_name, "config.pbtxt")
-    with open_file(config_path, "w") as f:
-        f.write(_get_model_config(model))
-    return model_path, config_path
+
+    device_types, instance_counts = get_device_types_and_counts(
+        preprocessor_num_instances, predictor_device_type, predictor_num_instances, postprocessor_num_instances
+    )
+    split_modules = get_inference_modules(model, device_types[1])
+    example_input = get_example_input(model, device_types, data_example)
+
+    paths = {}
+    triton_masters = []
+    for i, module in enumerate(split_modules):
+        example_input = place_on_device(example_input, device_types[i])
+        triton_master = TritonMaster(
+            module,
+            example_input,
+            INFERENCE_STAGES[i],
+            predictor_max_batch_size,
+            max_queue_delay_microseconds,
+            model_name,
+            output_path,
+            model_version,
+            model.config,
+            device_types[i],
+            model_instance_count=instance_counts[i],
```

Comment:
Should we use named parameters here for all args given the number of parameters, and the fact that order might change in future?